tests/integration/admin/test_admin_routes.py

```
@@ -108,6 +108,67 @@ def test_admin_inactive_coordinator_gets_403(self, mock_app, mock_client):
+        """Unauthenticated user should be redirected to login when trying to access DB admin."""
+        resp = mock_client.get("/adminpanel/db_admin/")
+        assert resp.status_code == 302
+        assert "login" in resp.headers["Location"]
+
+    def test_db_admin_requires_admin_role(self, mock_app, mock_client):
+        """A regular user (not coordinator/admin) should get 403 when trying to access DB admin."""
+        with mock_app.app_context():
+            uid = self._upsert_u_token(
+                username="RegularUser",
+                access_key="k",
+                access_secret="s",
```

Ruff (0.16.0)

[error] Possible hardcoded password assigned to argument: "access_secret"

(S106)